File: MG/App/Frontend/WEB_flutter/lib/models/modelsViews/userInfo.dart
class UserInfo {
  int id;
  String name;
  String lastname;
  String username;
  String password;
  String email;
  String phone;
  int cityId;
  int points;
  String cityName;
  String rankName;
  String profilePhoto;
  int reportCount;
  int postCount;
  int userTypeID;
  int isReported;
  bool isLogged;

  String get fullName => this.name + " " + this.lastname; //kada se kuca prvo ime pa prezime
  String get fullLastName => this.lastname + " " + this.name; //kada se kuca prvo prezime pa ime


  UserInfo(name, lastname, username, password, email, phone, cityId, points, cityName, rankName, reportCount, profilePhoto, postCount,  userTypeID, isReported, isLogged) {
    this.name = name;
    this.lastname = lastname;
    this.username = username;
    this.password = password;
    this.email = email;
    this.phone = phone;
    this.cityId = cityId;
    this.points = points;
    this.cityName = cityName;
    this.rankName = rankName;
    this.reportCount = reportCount;
    this.profilePhoto = profilePhoto;
	  this.postCount = postCount;
    this.userTypeID = userTypeID;
    this.isReported = isReported;
    this.isLogged = isLogged;  }

  Map<String, dynamic> toMap() {
    var map = Map<String, dynamic>();
    if (id != null) map['id'] = id;
    map['name'] = name;
    map['lastname'] = lastname;
    map['username'] = username;
    map['password'] = password;
    map['email'] = email;
    map['phone'] = phone;
    map['cityID'] = cityId;
    map['points'] = points;
    map['cityName'] = cityName;
    map['rankName'] = rankName;
    map['reportCount'] = reportCount;
    map['profilePhotoPath'] = profilePhoto;
  	map['postCount'] = postCount;
    map['userTypeID'] = userTypeID;
    map['isReported'] = isReported;
    map['isLogged'] = isLogged;
    return map;
  }

  UserInfo.fromObject(dynamic o) {
    this.id = o['id'];
    this.name = o['name'];
    this.lastname = o['lastname'];
    this.username = o['username'];
    this.password = o['password'];
    this.email = o['email'];
    this.phone = o['phone'];
    this.cityId = o['cityID'];
    this.points = o['points'];
    this.cityName = o['cityName'];
    this.rankName = o['rankName'];
    this.reportCount = o['reportCount'];
    this.profilePhoto = o['profilePhotoPath'];
	  this.postCount = o['postCount'];
    this.userTypeID = o['userTypeID'];
    this.isReported = o['isReported'];
    this.isLogged = o['isLogged'];
  }

}

File: MG/App/Backend/mojGradApp/mojGradApp/Models/InfoModels/UserInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace mojGradApp.Models.InfoModels
{
    public class UserInfo
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string? Lastname { get; set; }
        public string Username { get; set; }
        public string Password { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public int CityID { get; set; }
        public int Points { get; set; }
        public string ProfilePhotoPath { get; set; }
        public string? Address { get; set; }
        public int UserTypeID { get; set; }
        public bool? IsVerified { get; set; }

        public string CityName { get; set; }
        public string RankName { get; set; }
        public int ReportCount { get; set; }
        public int IsReported { get; set; }
        public int PostCount { get; set; }

        public UserInfo(UserDatas user)
        {
            this.Id = user.Id;
            this.Name = user.Name;
            this.Lastname = user.Lastname;
            this.Username = user.Username;
            this.Password = user.Password;
            this.Email = user.Email;
            this.Phone = user.Phone;
            this.CityID = user.CityID;
            int points = user.Point.Sum(x => x.Number);
            this.Points = points >=0 ? points : 0;
            this.Address = user.Address;
            this.CityName = user.City.Name;
            this.UserTypeID = user.UserTypeID;
            this.ProfilePhotoPath = user.ProfilePhotoPath;
            if (this.Points >= 300)
                this.RankName = "Zlato";
            else if (this.Points >= 200 && this.Points < 300)
                this.RankName = "Srebro";
            else if(this.Points >= 100 && this.Points < 200)
                this.RankName = "Bronza";
            else
                this.RankName = "Nema ranga";
            this.ReportCount = user.UsersThatAreReported.Count(x => x.ReportedUserID == this.Id);
            this.PostCount = user.Posts.Count(x => x.UserID == user.Id);
            this.IsVerified = user.IsVerified;
        }

        public UserInfo(UserDatas user, int userID) : this(user)
        {
            if (user.UsersThatAreReported.FirstOrDefault(x => x.ReportedUserID == user.Id && x.UserID == userID) != null)
                this.IsReported = 1;
            else
                this.IsReported = 0;
        }
    }
}
